File: app/src/main/java/com/example/breezil/favnews/repository/BookMarkRepository.java
package com.example.breezil.favnews.repository;

import android.app.Application;
import android.arch.lifecycle.LiveData;
import android.os.AsyncTask;

import com.example.breezil.favnews.db.AppDatabase;
import com.example.breezil.favnews.db.BookMarkDao;
import com.example.breezil.favnews.model.BookMark;

import java.util.List;

public class BookMarkRepository {
    private BookMarkDao bookMarkDao;
    private LiveData<List<BookMark>> allBookMarks;

    public BookMarkRepository(Application application){
        AppDatabase database = AppDatabase.getAppDatabase(application);
        bookMarkDao = database.bookMarkDao();
        allBookMarks = bookMarkDao.getAllBookMarks();


    }

    public void insert(BookMark bookMark){
        new InsertBookMark(bookMarkDao).execute(bookMark);
    }

    public void delete(BookMark bookMark){
        new DeleteBookMark(bookMarkDao).execute(bookMark);
    }

    public void deleteAllBookMark(){
        new DeleteAllBookMarks(bookMarkDao).execute();
    }

    public LiveData<List<BookMark>> getAllBookMarks() {
        return allBookMarks;
    }

    private static class InsertBookMark extends AsyncTask<BookMark, Void, Void>{
        private BookMarkDao bookMarkDao;
        public InsertBookMark(BookMarkDao bookMarkDao){
            this.bookMarkDao = bookMarkDao;
        }

        @Override
        protected Void doInBackground(BookMark... bookMarks) {
            bookMarkDao.insert(bookMarks[0]);
            return null;
        }
    }





    private static class DeleteBookMark extends AsyncTask<BookMark, Void, Void>{
        private BookMarkDao bookMarkDao;
        public DeleteBookMark(BookMarkDao bookMarkDao){
            this.bookMarkDao = bookMarkDao;
        }

        @Override
        protected Void doInBackground(BookMark... bookMarks) {
            bookMarkDao.delete(bookMarks[0]);
            return null;
        }
    }

    private static class DeleteAllBookMarks extends AsyncTask<Void,Void,Void>{
        private BookMarkDao bookMarkDao;

        public DeleteAllBookMarks(BookMarkDao bookMarkDao) {
            this.bookMarkDao = bookMarkDao;
        }

        @Override
        protected Void doInBackground(Void... voids) {
            bookMarkDao.deleteAllBookMarks();
            return null;
        }
    }

}

File: app/src/main/java/com/example/breezil/fave/repository/BookMarkRepository.kt
package com.example.breezil.fave.repository

import android.app.Application
import android.arch.lifecycle.LiveData
import android.os.AsyncTask

import com.example.breezil.fave.db.AppDatabase
import com.example.breezil.fave.db.BookMarkDao
import com.example.breezil.fave.model.BookMark

class BookMarkRepository(application: Application) {
    private val bookMarkDao: BookMarkDao
    val allBookMarks: LiveData<List<BookMark>>

    init {
        val database = AppDatabase.getAppDatabase(application)
        bookMarkDao = database.bookMarkDao()
        allBookMarks = bookMarkDao.allBookMarks


    }

    fun insert(bookMark: BookMark) {
        InsertBookMark(bookMarkDao).execute(bookMark)
    }

    fun delete(bookMark: BookMark) {
        DeleteBookMark(bookMarkDao).execute(bookMark)
    }

    fun deleteAllBookMark() {
        DeleteAllBookMarks(bookMarkDao).execute()
    }

    private class InsertBookMark(private val bookMarkDao: BookMarkDao) : AsyncTask<BookMark, Void, Void>() {

        override fun doInBackground(vararg bookMarks: BookMark): Void? {
            bookMarkDao.insert(bookMarks[0])
            return null
        }
    }


    private class DeleteBookMark(private val bookMarkDao: BookMarkDao) : AsyncTask<BookMark, Void, Void>() {

        override fun doInBackground(vararg bookMarks: BookMark): Void? {
            bookMarkDao.delete(bookMarks[0])
            return null
        }
    }

    private class DeleteAllBookMarks(private val bookMarkDao: BookMarkDao) : AsyncTask<Void, Void, Void>() {

        override fun doInBackground(vararg voids: Void): Void? {
            bookMarkDao.deleteAllBookMarks()
            return null
        }
    }

}
